src/main/java/com/hazelcast/remotecontroller/HazelcastCloudManager.java

```diff
@@ -0,0 +1,347 @@
+                LOG.warn("Body is null for tlsPassword");
+                return null;
+            }
+        }
+        catch(Exception e)
+        {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    private boolean waitForStateOfCluster(String clusterId, String expectedState, long timeoutInMillisecond) {
+        String currentState = "";
+        long startTime = System.currentTimeMillis();
+        int retryCycle = 10;
+        try {
+            while ((System.currentTimeMillis() - startTime) < timeoutInMillisecond) {
+                String query = String.format("{ \"query\": \"query { cluster(clusterId: \\\"%s\\\") { id state } }\" }", clusterId);
+                String responseBody = prepareAndSendRequest(query).body().string();
+                LOG.info(responseBody);
+                currentState = mapper.readTree(responseBody).get("data").get("cluster").get("state").asText();
+                if(currentState.equalsIgnoreCase(expectedState))
+                    return true;
+                if(currentState.equalsIgnoreCase("FAILED"))
+                {
+                    LOG.error("Cluster is FAILED");
+                    return false;
+                }
+                TimeUnit.SECONDS.sleep(retryCycle);
+            }
+            LOG.error(String.format("The cluster could not come to the given state in %d millisecond", timeoutInMillisecond));
+            return false;
+        } catch (Exception e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    private void waitForDeletedCluster(String clusterId, long timeoutInMillisecond) throws CloudException, InterruptedException {
+        long startTime = System.currentTimeMillis();
+        int retryCycle = 10;
+        while ((System.currentTimeMillis() - startTime) < timeoutInMillisecond) {
+            if (getHazelcastCloudCluster(clusterId) == null)
+                return;
+            TimeUnit.SECONDS.sleep(retryCycle);
+        }
+        throw new CloudException(String.format("The cluster is not deleted in %d millisecond", timeoutInMillisecond));
+    }
+
+    private String getBearerToken(String apiKey, String apiSecret) {
+        try {
+            String query = String.format("{ \"query\": \"mutation { login(apiKey: \\\"%s\\\", apiSecret: \\\"%s\\\") {token} }\" }", apiKey, apiSecret);
+            RequestBody body = RequestBody.create(JSON, query);
+            Request request = new Request.Builder()
+                    .url(HttpUrl.get(uri))
+                    .post(body)
+                    .header("Content-Type", "application/json")
+                    .build();
+            call = client.newCall(request);
+            Response response = call.execute();
+            String loginResponse = response.body().string();
+            LOG.info(maskValueOfToken(loginResponse));
+            return mapper.readTree(loginResponse).get("data").get("login").get("token").asText();
```

Comment:
Can you document what the response string looks like. You can also point to a web document that describes the reponse? Needed for maintenance.

Reply:
I did not see this fixed. By the way, for each REST call we need similar info or point to a document that identifies the query and responses.

Reply:
I added cloud API reference document to readme. It includes the endpoints